File: ToDoList/Back-End/ToDoList.Data/Dto/Assignment/AssignmentDto.cs
using ToDoList.Data.Dto.SubAssignment;

namespace ToDoList.Data.Dto.Assignment;

public class AssignmentDto
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public DateTime Deadline { get; set; }
    public bool Status { get; set; }
    public DateTime Reminder { get; set; }
    public ICollection<SubAssignmentDto> subAssignmentDto {get; set;}

}

File: ToDoList/Front-End/ToDoListFront/src/app/models/Assignment/AssignmentDto.ts
import { Guid } from "guid-typescript";
import { AssignmentListDto } from "../AssignmentList/assignmentListDto";

export class AssignmentDto{
    public id?: Guid;
    public name?: string;
    public deadline?: Date;
    public status?: boolean;
    public assignmentList?: AssignmentListDto[];
}